### PR TITLE
quickcache slow things on page load

### PR DESCRIPTION
**Pull request by dannyroberts**
takes 150-175ms off for me locally (on dummy page that just extends 2-column layout)

### CODE REVIEW

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 1]
```diff
@@ -135,13 +146,11 @@ def domains_for_user(context, request, selected_domain=None):
     Cache the entire string alongside the couch_user's doc_id that can get invalidated when
     the user doc updates via save.
     """
-    domain_list = []
+
     if selected_domain != 'public':
```
Comment (czue) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: we can kill this. the public domain is dead

File: corehq/apps/domain/utils.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def get_domain_module_map():
     return hardcoded
 
 
+@quickcache(['domain'])
```
Comment (czue) [inline corehq/apps/domain/utils.py]: :+1:

**Comment by czue** [comment]
seems great! i guess there may be some blips where your list isn't accurate for a bit if you are added/removed from a domain? wasn't sure if you were thinking of adding a skip or something or we think that's fine

**Comment by dannyroberts** [comment]
Yeah that's a good point. There are other things cached upstream from this that get cleared on user save (I believe the source of the slowness is that what `Domain.active_for_couch_user` caches full Domain objects which are pretty bloated, when all you need is the list of domain names and their URLs); since I think those have dealt with cache invalidation, I'll just hook into the same thing.

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by czue** [comment]
looks good - being paranoid and not merging due to label, but ready to go from my side.

**Comment by dannyroberts** [comment]
Ok, just did a final manual check on the invalidation (local runserver). I'll go ahead and merge

**Pull request merged**